src/sources/host_metrics.rs

```diff
@@ -692,6 +694,22 @@ fn add_collector(collector: &str, mut metrics: Vec<Metric>) -> Vec<Metric> {
     metrics
 }
 
+#[cfg(target_os = "linux")]
+pub fn maybe_set_roots() {
```

Comment:
I suggest something like `init_roots`, and call with `host_metrics::init_roots` in `app.rs`.